### PR TITLE
[Github] Improvements to My Issues and My Pull Requests commands and Menu Bar

### PR DESCRIPTION
**Pull request by jfkisafk**
## Description

- My Issues Command + Menu Bar: Preferences to specify categories to display. Issues opened by the user will always be displayed.
- My PRs Command + Menu Bar: Preferences to specify categories to display. PRs created by the user will always be displayed.
- The categories not being displayed will not invoke extra API calls, reducing throttling.
- ⚠️My PRs Menu Bar: Removed `includeCount` options to keep things simple (and in-sync) by just displaying the count of PRs in the categories selected.

#### My Pull Requests

Screenshot 2024-08-07 at 2 37 27 AM
Screenshot 2024-08-07 at 2 38 35 AM

#### My Issues

Screenshot 2024-08-07 at 2 40 27 AM
Screenshot 2024-08-07 at 2 40 46 AM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

* closes https://github.com/raycast/extensions/issues/11626
* closes https://github.com/raycast/extensions/issues/12664

### CODE REVIEW

File: extensions/github/package.json [hunk 1]
```diff
@@ -407,7 +445,7 @@
     "@parcel/watcher": "^2.4.1",
     "@raycast/eslint-config": "^1.0.8",
     "@types/lodash": "^4.17.5",
-    "@types/node": "^20.14.5",
```
Comment (LitoMore) [inline extensions/github/package.json:410]: You should not use 22 since Raycast is always using LTS. The current LTS is 20.

**Review by LitoMore — CHANGES_REQUESTED** [review]

File: extensions/github/package.json (hunk 1 above)
Comment (jfkisafk) [reply extensions/github/package.json:410]: Hmm this is just `devDependencies` and I usually do `npm i @types/node@ts5.5 -D` to keep it consistent with the typescript version. In production, raycast will use node@lts (20) anyways. Plus, we have the preference to use production node in development mode. I think this is a non-issue and being used in other commands too: [search](https://github.com/search?q=repo%3Araycast%2Fextensions+%40types%2Fnode%5C%22%3A+%2F%22%5C%5E22%2F&type=code).

Comment (LitoMore) [reply extensions/github/package.json:410]: I don't think so. Some Node.js APIs are different in 22, which will confuse developers.

Comment (jfkisafk) [reply extensions/github/package.json:410]: Just curious, what API changes are we concerned about here? [All changes](https://nodejs.org/en/blog/announcements/v22-release-announce) I could see do not impact raycast development (especially when you turn on production mode). I can revert the version, but given v22 will enter LTS in October, I'd like to understand if this is a blocking change? If it is, I can create a tangent PR to move all extensions using v22 back to v20.

Comment (jfkisafk) [reply extensions/github/package.json:410]: If this is not blocking, then I would prefer to future-proof with this change, or revert if any other changes are required

Comment (LitoMore) [reply extensions/github/package.json:410]: This is more of your own personal preference, which leaves a potential compatibility issue for the project, and you want others to help you clean up the mess after the problem, which is not appropriate.

This is an open-source project, and we should try to include as few personal preferences as possible without practical significance.

Comment (j3lte) [reply extensions/github/package.json:410]: I'm with @jfkisafk on this one. Sure, some Node.JS apis are different, but we're not really using them. Only if you are really using Node specific APIs like `fs` it **might** break (although the diffe. That's not the case here...

Comment (jfkisafk) [reply extensions/github/package.json:410]: Thanks for raising the bar guys! Based on the discussion, I will resolve this thread as this is a non-blocking issue. As suggested above, if `@types/node@22` shows incompatibility, I will personally revert all extensions back to v20. I am not expecting others to clean it up!

**Comment by jfkisafk** [comment]
Hello @thomaslombart can you please take a look if you find some time? 🙏

This should help reduce transient throttling users are seeing with these github commands.

**Pull request merged**